src/sonic-config-engine/sonic_platform.py

```diff
@@ -41,12 +41,18 @@ def get_sonic_version_info():
     return data
 
 def get_system_mac():
-    proc = subprocess.Popen("ip link show eth0 | grep ether | awk '{print $2}'", shell=True, stdout=subprocess.PIPE)
+    version_info = get_sonic_version_info()
+
+    if (version_info['asic_type'] == 'mellanox'):
+        get_mac_cmd = "sudo decode-syseeprom -m"
+    else:
+        get_mac_cmd = "ip link show eth0 | grep ether | awk '{print $2}'"
```

Comment:
read a file "/sys/class/net/eth0/address"?

Reply:
like above - not sure about other platforms